diff --git a/handlers.go b/handlers.go
index 5c5ab09..bd2faa4 100644
--- a/handlers.go
+++ b/handlers.go
@@ -131,21 +131,23 @@ func (r *oauthProxy) oauthCallbackHandler(w http.ResponseWriter, req *http.Reque
 		r.log.Error("unable to parse id token for identity", zap.Error(err))
 		r.accessForbidden(w, req)
 		return
+	} else if err = verifyToken(r.client, token); err != nil {
+		r.log.Error("unable to verify the id token", zap.Error(err))
+		r.accessForbidden(w, req)
+		return
 	}
+
+	// step: check the access token is valid
 	access, id, err := parseToken(resp.AccessToken)
-	if err == nil {
+	if err != nil {
+		r.log.Warn("unable to parse the access token, using id token only", zap.Error(err))
+    } else if err = verifyToken(r.client, access); err != nil {
+		r.log.Warn("unable to verify the access token, using id token only", zap.Error(err))
+	} else {
 		token = access
 		identity = id
-	} else {
-		r.log.Warn("unable to parse the access token, using id token only", zap.Error(err))
 	}
 
-	// step: check the access token is valid
-	if err = verifyToken(r.client, token); err != nil {
-		r.log.Error("unable to verify the id token", zap.Error(err))
-		r.accessForbidden(w, req)
-		return
-	}
 	accessToken := token.Encode()
 
 	// step: are we encrypting the access token?
