diff --git a/vendor/github.com/gambol99/go-oidc/oauth2/oauth2.go b/vendor/github.com/gambol99/go-oidc/oauth2/oauth2.go
index fdb69ab..615ae8c 100644
--- a/vendor/github.com/gambol99/go-oidc/oauth2/oauth2.go
+++ b/vendor/github.com/gambol99/go-oidc/oauth2/oauth2.go
@@ -257,7 +257,6 @@ func (c *Client) RequestToken(grantType, value string) (result TokenResponse, er
 	v := c.commonURLValues()
 
 	v.Set("grant_type", grantType)
-	v.Set("client_secret", c.creds.Secret)
 	switch grantType {
 	case GrantTypeAuthCode:
 		v.Set("code", value)
